scalafmt-core/shared/src/main/scala/org/scalafmt/Scalafmt.scala

```diff
@@ -101,14 +105,34 @@ object Scalafmt {
       file: String,
       range: Set[Range] = Set.empty
   ): Try[String] =
-    if (!FileOps.isAmmonite(file)) doFormatOne(code, style, file, range)
-    else {
+    if (FileOps.isAmmonite(file)) {
       // XXX: we won't support ranges as we don't keep track of lines
       val chunks = ammonitePattern.split(code)
       if (chunks.length <= 1) doFormatOne(code, style, file, range)
       else
         flatMapAll(chunks.iterator)(doFormatOne(_, style, file))
           .map(_.mkString("\n@\n"))
+    } else if (FileOps.isMarkdown(file)) {
+      if (code.contains("```scala mdoc")) {
+        val markdown = MarkdownFile.parse(Input.VirtualFile(file, code))
+        flatMapAll(markdown.parts.iterator) {
+          case text: Text => Success(text.value)
+          case fence: CodeFence =>
+            if (fence.info.value.startsWith("scala mdoc")) {
+              doFormatOne(fence.body.value, style, file).map { formatted =>
+                fence.newBody = Some(formatted.trim)
+              }
+            } else {
+              Success("")
+            }
+        }.map { _ =>
+          markdown.renderToString
+        }
```

Comment:
you don't need `flatMapAll`. why not

```
  val resultIterator = markdown.parts.iterator.collect {
    case fence: CodeFence if fence.info.value.startsWith("scala mdoc") =>
      val res = doFormatOne(fence.body.value, style, file)
      res.foreach { formatted =>
        // XXX if you need to trim, why not trim fence.body before formatting?
        fence.newBody = Some(formatted.trim)
      }
      res
  }
  if (resultIterator.isEmpty) Success(code)
  else
    resultIterator
      .collectFirst { case f: Failure[_] => f }
      .getOrElse(Success(markdown.renderToString))
```

Reply:
Thanks! Using this now.
I looked into pre-trimming per your note, but it ends up leaving an extra blank line at the end of the output and failing the current tests.